contracts/registries/LicenseNFT.sol

```diff
@@ -0,0 +1,280 @@
+    event BatchMetadataUpdate(uint256 _fromTokenId, uint256 _toTokenId);
+
+    /// @dev Storage of the LicenseNFT
+    /// @custom:storage-location erc7201:story-protocol.LicenseNFT
+    struct LicenseNFTStorage {
+        string imageUrl;
+        ILicensingModule licensingModule;
+        IDisputeModule disputeModule;
+        uint256 totalMintedTokens;
+        mapping(uint256 tokenId => LicenseTokenMetadata) licenseTokenMetadatas;
+    }
+
+    // TODO: update the storage location
+    // keccak256(abi.encode(uint256(keccak256("story-protocol.LicenseNFT")) - 1)) & ~bytes32(uint256(0xff));
+    bytes32 private constant LicenseNFTStorageLocation =
+        0x5ed898e10dedf257f39672a55146f3fecade9da16f4ff022557924a10d60a900;
+
+    modifier onlyLicensingModule() {
+        if (msg.sender != address(_getLicenseNFTStorage().licensingModule)) {
+            revert Errors.LicenseNFT__CallerNotLicensingModule();
+        }
+        _;
+    }
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(address governance, string memory imageUrl) public initializer {
+        __ERC721_init("Programmable IP License NFT", "PILNFT");
+        __GovernableUpgradeable_init(governance);
+        __UUPSUpgradeable_init();
+        _getLicenseNFTStorage().imageUrl = imageUrl;
+    }
+
+    /// @dev Sets the LicensingModule address.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param newLicensingModule The address of the LicensingModule
+    function setLicensingModule(address newLicensingModule) external onlyProtocolAdmin {
+        if (newLicensingModule == address(0)) {
+            revert Errors.LicenseNFT__ZeroLicensingModule();
+        }
+        LicenseNFTStorage storage $ = _getLicenseNFTStorage();
+        $.licensingModule = ILicensingModule(newLicensingModule);
+    }
+
+    function setDisputeModule(address newDisputeModule) external onlyProtocolAdmin {
+        if (newDisputeModule == address(0)) {
+            revert Errors.LicenseNFT__ZeroDisputeModule();
+        }
+        LicenseNFTStorage storage $ = _getLicenseNFTStorage();
+        $.disputeModule = IDisputeModule(newDisputeModule);
+    }
+
+    /// @dev Sets the Licensing Image URL.
+    /// @dev Enforced to be only callable by the protocol admin
+    /// @param url The URL of the Licensing Image
+    function setLicensingImageUrl(string calldata url) external onlyProtocolAdmin {
+        LicenseNFTStorage storage $ = _getLicenseNFTStorage();
+        $.imageUrl = url;
+        emit BatchMetadataUpdate(1, $.totalMintedTokens);
+    }
+
+    function mintLicenseTokens(
+        address originalIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        uint256 amount, // mint amount
+        address minter,
+        address receiver
+    ) external onlyLicensingModule returns (uint256 startLicenseTokenId, uint256 endLicenseTokenId) {
```

Comment:
It is an easy-of-use feature; the caller can certainly be sure that a range of token IDs was generated instead of assuming from the `amount`. but as there is no specific usage, I agree we can drop it for now.